### ISSUE TITLE
Occasional segfaults on macOS Sierra (10.12)

### ISSUE THREAD

**Issue by codykrieger** [github/hub#1257]
I upgraded to the GM build of macOS Sierra today, and with hub 2.2.5 (installed via homebrew), I'm seeing occasional segfaults when doing pretty standard git operations. I've seen this behavior the most when deleting branches (i.e. `git branch -d foo`).

The crash logs don't look suuuuuper useful, but I've attached one nonetheless (in .zip format, because GitHub): [hub_2016-09-08-002844_gravity-mk2.crash.zip](https://github.com/github/hub/files/461133/hub_2016-09-08-002844_gravity-mk2.crash.zip).

Relevant bit:
```
Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   hub                           	0x000000000000faeb runtime.unlock + 331 (lock_sema.go:107)
```

(Note that I have `alias git=hub` in my shell (zsh) config.)

**Comment by mislav** [comment]
Thanks for reporting! Go versions prior to 1.6.3 aren't compatible with Sierra. I wonder how we can find out which Go version did Homebrew use to compile hub for you?

**Comment by codykrieger** [comment]
@mislav Oh, interesting. I didn't know that.

It looks like the version of `hub` I've got was compiled with 1.6.3:

```
% strings `which hub` | grep 'go1\.'
go1.6.3
```

I'll reinstall and compile with 1.7.1 and see if I can reproduce the crash.

**Comment by codykrieger** [comment]
I ran `git checkout -b foo && git checkout master && git branch -d foo` about 100 times without reproducing the crash, so it seems to be fixed now. 👌

FWIW, the version of `hub` I had installed (the one compiled with 1.6.3) was one of those pre-built “bottles” that homebrew installs by default, so maybe it'd be worth having that pre-built copy recompiled with 1.7.1? Perhaps that's an issue for the homebrew maintainers to take up, since I think they're responsible for all of the pre-built packages?

Anyway, thanks for the quick response! I'll close this.
